### PR TITLE
Rust UI: ask_pagination support in confirm_blob

### LINKED ISSUE #2557: `confirm_blob` doesn't support `ask_pagination` yet (bitcoin doesn't need it)

### PR DESCRIPTION
**Pull request by mmilata**
Depends on #2519. Closes #2557. [[UI diff](https://satoshilabs.gitlab.io/-/trezor/trezor-firmware/-/jobs/3185069287/artifacts/master_diff/index.html)]

emu00000000 emu00000001 emu00000003

### CODE REVIEW

**Review by grdddj — COMMENTED** [review]
I like it, and users will surely do as well! 

For a short moment, I was not fond of `num_traits::FromPrimitive`, as the `Font::from_i32` functionality is not defined directly, but it really saves effort over implementing and maintaining the same functionality ourselves. (And it is impossible to implement `From<i32> for Option<Font>`)

Also, first time looking at 
image
I was thinking the button is swippable to the right. :) That is most probably not a problem, theoretically, we could change the wording to `SWIPE UP TO CONTINUE`, but that doesn't fit properly with the current font size:
image

**Comment by mmilata** [comment]
Rebased to master [[UI diff](https://satoshilabs.gitlab.io/-/trezor/trezor-firmware/-/jobs/3218851538/artifacts/master_diff/index.html)].

Regarding the swipe-button confusion we can raise this with the design team, though I believe the UI1 design this is based on came from them. Fortunately side swipe does nothing here, and hopefully by the time a user sees this screen they've used a button or a scrollable screen before?

**Comment by grdddj** [comment]
> Fortunately side swipe does nothing here, and hopefully by the time a user sees this screen they've used a button or a scrollable screen before?

Yes, I agree this is a probable case, users will already know what to do most of the time. And also they cannot do any harm just experimenting.

File: core/embed/rust/src/ui/model_tt/layout.rs [hunk 1]
```diff
@@ -751,21 +755,58 @@ extern "C" fn new_confirm_with_info(n_args: usize, args: *const Obj, kwargs: *mu
...
 
-        let mut paragraphs = ParagraphVecLong::new();
+        let mut paragraphs = ParagraphVecShort::new();
 
         let mut iter_buf = IterBuf::new();
         let iter = Iter::try_from_obj_with_buf(items, &mut iter_buf)?;
-        for text in iter {
+        for para in iter {
+            let [font, text]: [Obj; 2] = iter_into_objs(para)?;
+            let style: &TextStyle = theme::textstyle_number(font.try_into()?);
```
[the remark this answers is not in the document]
Comment (mmilata) [reply core/embed/rust/src/ui/model_tt/layout.rs:776]: agreed, but the ethereum structured data signing currently uses it and will have to be changed

**Pull request merged**